### PR TITLE
[Excel] (Data types) Add preview sample for web images

### PR DESCRIPTION
**Pull request by alison-mk**
[Internal work item.](https://office.visualstudio.com/OC/_workitems/edit/5763605)

@AlexJerabek I previously mentioned that I would include a strategy to update multiple cells in this web image data types sample, but after research I think that the formatted number value sample is better suited to showcasing how to update multiple cells. So, my plan is to update the FNV sample instead, after this web image sample is finalized.

### CODE REVIEW

File: samples/excel/85-preview-apis/data-types-web-image.yaml
Comment (AlexJerabek) [inline samples/excel/85-preview-apis/data-types-web-image.yaml:40]: Consider adding a /* comment */ explaining why "Error" is used here. Is it a best practice for types that have no basic value equivalent?

Comment (AlexJerabek) [inline samples/excel/85-preview-apis/data-types-web-image.yaml:40]: Can we use the enum instead of the string variant? I think the enums help teach the code better by showing the possibilities (instead of making it look like arbitrary strings can go here).

File: samples/excel/85-preview-apis/data-types-web-image.yaml
Comment (AlexJerabek) [inline samples/excel/85-preview-apis/data-types-web-image.yaml:36]: Instead of directly assigning the JSON, consider creating a WebImageCellValue variable as an intermediary step. That way, the type is clear and readers in the ref docs can know what to look up.

File: samples/excel/85-preview-apis/data-types-web-image.yaml
Comment (AlexJerabek) [inline samples/excel/85-preview-apis/data-types-web-image.yaml:93]: Consider shortening this a little.
```suggestion
              console.log("The selected cell is missing an image URL. Select a cell that contains an image.");
```

File: snippet-extractor-output/snippets.yaml
Comment (AlexJerabek) [inline snippet-extractor-output/snippets.yaml:5132]: Is there value in having one of these two samples on the same page map to the `openImage` function instead? That way, you're showing both get and set functionality.

**Review by AlexJerabek — APPROVED** [review]
Approving with several suggestions.  I'm happy to take another look in the changes end up being more work.

File: samples/excel/85-preview-apis/data-types-web-image.yaml
Comment (alison-mk) [reply samples/excel/85-preview-apis/data-types-web-image.yaml:40]: Good point. Yes, it's a best practice.

**Review by alison-mk — COMMENTED** [review]
Thanks for the review! Responding with a few questions.

File: samples/excel/85-preview-apis/data-types-web-image.yaml
Comment (AlexJerabek) [reply samples/excel/85-preview-apis/data-types-web-image.yaml:142]: There's a `.github/images` folder in this repo. Maybe putting an image there and referencing the published URL would be good (you'd have to add and test that separately). I don't think the current solution is bad, by any means, just thinking of ways to remove user friction.

File: samples/excel/85-preview-apis/data-types-web-image.yaml
Comment (AlexJerabek) [reply samples/excel/85-preview-apis/data-types-web-image.yaml:156]: I mean the scenario where the user adds and image and clicks through the rest of the buttons. I didn't notice the fields change when first testing this.

File: samples/excel/85-preview-apis/data-types-web-image.yaml
Comment (AlexJerabek) [reply samples/excel/85-preview-apis/data-types-web-image.yaml:40]: I'm just biased against that string overload... I think the enum is easier to read and maintain from a user perspective. Your call either way.

Comment (alison-mk) [reply samples/excel/85-preview-apis/data-types-web-image.yaml]: I misspoke. `basicType` and `basicValue` are readonly properties. They can technically be written, but they have to be written as the values shown here. To simplify and avoid confusion, I will remove these two properties from the Script Lab sample.

File: samples/excel/85-preview-apis/data-types-web-image.yaml
Comment (alison-mk) [reply samples/excel/85-preview-apis/data-types-web-image.yaml:153]: I've updated the description before this button to better explain where the image details will display. I think it's sufficient, but if you feel strongly (or if we get feedback from customers) then I can adjust in the future.

File: samples/excel/85-preview-apis/data-types-web-image.yaml
Comment (alison-mk) [reply samples/excel/85-preview-apis/data-types-web-image.yaml:139]: I'll leave the image as-is for now, but if we get customer feedback that it's unclear then I will incorporate this suggestion.

**Comment by alison-mk** [comment]
@AlexJerabek Thanks for the review and responses to my questions! I didn't make substantial changes to the sample (and provided explanations why not), so I'll merge instead of requesting that you take a second look.

**Pull request merged**
